### PR TITLE
Complete the doc comments for filters

### PR DESCRIPTION
**Pull request by dougbu**
- #3851
- add lots of words
- correct that vs. which confusion

### CODE REVIEW

**Comment by dougbu** [comment]
Note this will likely go into RC2 but the PR is against `dev` for now.

**Comment by dougbu** [comment]
FYI this is approved for RC2. I'll port to `release` branch after review is complete.

File: src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/IFilterProvider.cs [hunk 1]
```diff
@@ -26,8 +29,16 @@ public interface IFilterProvider
         /// </remarks>
         int Order { get; }
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/IFilterProvider.cs]: Ima let you finish. But these remarks are some of the best remarks of all time :sunglasses:

File: src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/IFilterProvider.cs [hunk 2]
```diff
@@ -3,6 +3,9 @@
 
 namespace Microsoft.AspNetCore.Mvc.Filters
 {
+    /// <summary>
+    /// A <see cref="FilterItem"/> provider.
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/IFilterProvider.cs]: I'd mention here that a filter provider can "provide an executable filter associated with an IFilterMetadata by setting FilterItem.Filter, or add FilterItems to add new filters"

File: src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs [hunk 3]
```diff
@@ -13,12 +13,13 @@
 namespace Microsoft.AspNetCore.Mvc.Authorization
 {
     /// <summary>
-    /// An implementation of <see cref="IAsyncAuthorizationFilter"/>
+    /// An implementation of <see cref="IAsyncAuthorizationFilter"/> which applies a specific
+    /// <see cref="AuthorizationPolicy"/>.
     /// </summary>
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs]: I would add somewhere here that this is the filter that's added by the framework where you use `[Authorize(...)]`

File: src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs [hunk 4]
```diff
@@ -9,12 +9,20 @@
 
 namespace Microsoft.AspNetCore.Mvc
 {
+    /// <summary>
+    /// A filter that activates another filter of type <see cref="ImplementationType"/>.
+    /// </summary>
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs]: This needs some kinds of description of 'activates'.

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs (hunk 4 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs]: @rynowak we use "activates" in a few other places e.g. `Controller.ViewData` is `activated when <see cref="Controllers.IControllerActivator"/> activates`. Do those other uses (at least those outside `IRazorPageActivator` and similar) need more words? What words are you suggesting?

Without a specific suggestion, I'll switch to "creates" here.

**Comment by dougbu** [comment]
:up::date:
@rynowak please have a quick look at the words I added to address your comments.

File: src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs (hunk 4 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/TypeFilterAttribute.cs]: Specifically that it will retrieve the constructor parameters from DI

**Comment by rynowak** [comment]
updates looks good :shipit: x2

**Comment by dougbu** [comment]
eda647e into `release`
Also merged into `dev`

**Pull request closed without merging**